liferay-tango/liferay-portal
# LPS-124236 As a content author, I want to be able to export data to a excel file

### Motivation
The content dashboard is the only place in the portal where content managers can see all their web contents at once, regardless of the site or asset library it is stored or published in.

Currently there isn’t a way to export the list of contents in an easy way, and it is something that we’ve been asked for by customers and users.

This story tries to cover the MVP for downloading the list of contents without selecting columns, metadata or dates.

### Use cases
A content marketing team wants to run a content audit out of the portal or using an external tool.
A content marketing team wants to plan their publications and so they need the list of available assets.(both use cases related to this spreadsheet of our content mkt team: !https://developers.google.com/drive/images/drive_icon.png!Content Creation Matrix )
A company wants to upgrade / migrate and they need a way to control or audit the existing content.

### Objectives
Users must be able to download a .xls file with all items in the list (considering filters applied)
The ..xls file will include at least all the visible columns in the list.
Users won’t be able to select columns or other metadata at the moment.

### How to test
1. Need some content in the Content Dashboard, if total content is 0, the button is disabled
2. Click on Download btn
3. A loading state should appear and a cancel btn. If the total of documents is low this process could be very fast and this state can be barely visible.
4. If the cancel btn is clicked, the request should be canceled and the user sees a warning with the proper message
5. If the user tries to navigate while the generation is in process, a modal preventing direct navigation should pop up, both options are provided, to stay or cancel request and leave.
6. If the user reloads the page, a native alert should pop up.
7. One the process is completed, the file should be in the user computer, and the btn should provide feedback for 2 seconds, after that the btn should return to the initial state,
